front-end/hot-dog/src/CustomerComponents/TableRow.js

```diff
@@ -3,7 +3,27 @@ import React from 'react';
 class TableRow extends React.Component{
     constructor(props){
         super(props);
-        this.state = {}
+        this.state = {Quantity:0}
+    }
+
+    componentDidUpdate(){
+        if(this.props.quantity !== this.state.Quantity){
+            this.props.onChange({ITEM_ID:this.props.item, quantity:this.state.Quantity})
+        }
+    }
+
+    increaseQuantity = () =>{
+        let newValue = this.state.Quantity + 1
+        this.setState({Quantity:newValue})
+    }
+
+    decreaseQuantity = () =>{
+        let newValue = this.state.Quantity - 1
+        if(newValue < 0){
+            this.setState({Quantity:0})
```

Comment:
Not sure if I'm correct so take it with a grain of salt. setting state when it's already 0 would still trigger a re-render, so the "better" way may be to just ignore changes (do nothing) when the newValue is < 0.